glambie/util/timeseries_helpers.py

```diff
@@ -593,3 +596,115 @@ def get_average_trends_over_new_time_periods(start_dates, end_dates, changes, ne
+    Linear interpolation is performed on elevation_change_per_day values, to fill gaps in the input timeseries. An
+    overall elevation change in the data gap is then added to the original dataframe.
+
+    Parameters
+    ----------
+    timeseries_dataframe : _type_
+        _description_
+    """
+    date_gaps = [datetime.datetime.strptime(input_dataframe.start_date[i + 1], '%d/%m/%Y') - datetime.datetime.strptime(
+        input_dataframe.end_date[i], '%d/%m/%Y') for i in range(len(input_dataframe) - 1)]
+    date_gaps_in_days = [a.days for a in date_gaps]
+    input_dataframe['date_gap_days'] = np.append(date_gaps_in_days, [0])
+
+    start_dates = [datetime.datetime.strptime(a, '%d/%m/%Y') for a in input_dataframe.start_date]
+    end_dates = [datetime.datetime.strptime(a, '%d/%m/%Y') for a in input_dataframe.end_date]
+    new_start_dates, new_end_dates, new_changes, new_errors, new_hydro, new_sea_level = [], [], [], [], [], []
+
+    for i in range(len(input_dataframe.start_date) - 1):
+        current_start_date = start_dates[i]
+        current_end_date = end_dates[i]
+        new_start_dates.append(current_start_date)
+        new_end_dates.append(current_end_date)
+        new_changes.append(input_dataframe.glacier_change_observed[i])
+        new_errors.append(input_dataframe.glacier_change_uncertainty[i])
+        new_hydro.append(input_dataframe.hydrological_correction_value[i])
+        new_sea_level.append(input_dataframe.sea_level_correction_value[i])
+
+        if date_gaps_in_days[i] > 1:
+            gap_start_date = current_end_date + relativedelta(days=1)
+            gap_end_date = start_dates[i + 1] - relativedelta(days=1)
+            new_start_dates.append(gap_start_date)
+            new_end_dates.append(gap_end_date)
+            new_changes.append(np.nan)
+            new_errors.append(np.nan)
+            new_hydro.append(np.nan)
+            new_sea_level.append(np.nan)
+
+    # Add last row in
+    new_start_dates.append(start_dates[-1])
+    new_end_dates.append(end_dates[-1])
+    new_changes.append(input_dataframe.glacier_change_observed.tolist()[-1])
+    new_errors.append(input_dataframe.glacier_change_uncertainty.tolist()[-1])
+    new_hydro.append(input_dataframe.hydrological_correction_value.tolist()[-1])
+    new_sea_level.append(input_dataframe.sea_level_correction_value.tolist()[-1])
+
+    interpolated_dataframe = pd.DataFrame()
+    interpolated_dataframe['start_date'] = [datetime.datetime.strftime(a, '%d/%m/%Y') for a in new_start_dates]
+    interpolated_dataframe['end_date'] = [datetime.datetime.strftime(a, '%d/%m/%Y') for a in new_end_dates]
+    interpolated_dataframe['glacier_change_observed'] = new_changes
+    interpolated_dataframe['glacier_change_uncertainty'] = new_errors
+    interpolated_dataframe['hydrological_correction_value'] = new_hydro
+    interpolated_dataframe['sea_level_correction_value'] = new_sea_level
```

Comment:
I've decided not to implement it like this, because of the other columns in the submitted glambie datasets that aren't dealt with by the interpolate_change_per_day_to_fill_gaps() function. Instead they are currently added back into the interpolated dataframe in fix_non_grace_gravimetry_gaps() - and before each extra column is filled in, we need to check if it contains any non-identical values (in which case we can't do a simple fill and will need to consider this file individually). I don't want to include this functionality in the interpolate_change_per_day_to_fill_gaps() function, as livia wanted to be able to use it with dataframes that do just have the 6 columns it deals with. So I don't want to make a copy of the input dataframe with all of its columns, instead just creatingthe 6 that need interpolating, to keep things more general.  Sorry if this isn't a great explanation, happy to chat about it!

Reply:
I think I've understood and I've updated my suggestion for your use case, but correct me if I'm wrong and it doesnt work!

Reply:
Reading through this now and not at the very end of the day, this makes complete sense and is an obvious fix to my comment above :) Have made this change!